src/Input/Select/Select.noTypeChecking.test.tsx

```diff
@@ -9,35 +9,50 @@ import userEvent from '@testing-library/user-event';
 
 import Select from './Select';
 
-describe('<Select /> can still work when receives non-string option children or value', () => {
-  test('non-string option children, click option', async () => {
+describe('<Select /> can still work when receives non-string and non-number option children or value', () => {
+  test('non-string and non-number option children, click option', async () => {
     const onChange = jest.fn();
 
     const Component = () => (
-      <Select label="non-string option children" onChange={onChange}>
-        <Select.Option value="one">{1}</Select.Option>
-        <Select.Option value="two">{2}</Select.Option>
-        <Select.Option value="three">{3}</Select.Option>
+      <Select
+        label="non-string and non-number option children"
+        onChange={onChange}
+      >
+        <Select.Option value="one">
+          <span data-testid="option-1">1</span>
+        </Select.Option>
+        <Select.Option value="two">
+          <span data-testid="option-2">2</span>
+        </Select.Option>
+        <Select.Option value="three">
+          <span data-testid="option-3">3</span>
+        </Select.Option>
       </Select>
     );
 
     const { queryByText, asFragment } = render(<Component />);
-
     const downArrowIcon = document.querySelector('.select-icon');
     fireEvent.click(downArrowIcon);
     const option1 = queryByText('1');
     fireEvent.mouseDown(option1);
 
-    expect(onChange).toHaveBeenLastCalledWith('one');
+    expect(onChange).toHaveBeenCalledTimes(1);
+
     expect(asFragment()).toMatchSnapshot();
   });
 
-  test('non-string option children, search filtering', async () => {
+  test('non-string and non-number option children, search filtering', async () => {
     const Component = () => (
-      <Select label="non-string option children">
-        <Select.Option value="one">{1}</Select.Option>
-        <Select.Option value="two">{2}</Select.Option>
-        <Select.Option value="three">{3}</Select.Option>
+      <Select label="non-string and non-number option children">
+        <Select.Option value="one">
+          <span>1</span>
```

Comment:
i think you forgot to remove the span blocks here

Reply:
Oh no, it is intended. I want to add a component inside the option